### DISCUSSION TITLE
blog/adguard-home-tailscale-erase-ads-on-the-go/

### DISCUSSION THREAD

**Discussion by giscus** [AkashRajpurohit/AkashRajpurohit#40] [Comments]
# blog/adguard-home-tailscale-erase-ads-on-the-go/

Fed up with pesky online ads? In this blog, I'll show you how a dynamic duo, AdGuard Home and Tailscale, can give you ad-free browsing anytime, anywhere. It's a technical adventure that's worth every click.

https://akashrajpurohit.com/blog/adguard-home-tailscale-erase-ads-on-the-go/

**Comment by catalans1980** [comment]
Hi, thanks a lot for the tutorial. I'm trying to do this but with an Adguard on a docker container on my Synology NAS. The Adguard works perfectly in my local network, but i'm having no luck using it as DNS server remotelly. So far, Tailscale works perfectly: I can connect remotely to my NAS for PC outside my LAN, from my smartphone, etc. I even have the NAS as exit node and subnets published (and accepting outbound traffic). I'm able to Rsync the NAS to my other home PC drive. But i think I'm stuck at some config with the adguard... I have configured tailscale with the global nameservers writting there my NAS tailscale IP (100.x.x.x.x), but neither the DNS rewrites work, or i see DNS queries from out the LAN. Any clue? I think it might be the  "--accept-dns=false", I don't know how to apply that to my Tailscale installed on the NAS. THanks a ton!!!

**Reply by AkashRajpurohit** [reply]
> I think it might be the "--accept-dns=false"

Yes, you would need to do this as well as disable the magic DNS so that you can use the DNS from AdGuard home.

**Reply by camemb3rt** [reply]
- On the Tailscale App, go into Settings (your profile icon) > DNS Settings > Turn off "Use Tailscale DNS"

**Reply by camemb3rt** [reply]
Actually, messing around with it a bit more, try using the local IP address (192.x.x.x) as the global nameserver in the tailscale admin panel instead of the Tailscale IP (100.x.x.x), now DNS rewrites also work.

**Comment by jelbo** [comment]
It won't work for me either. I have done all of this:
- Advertise subnet routes so I can reach AGH on it's LAN IP
- Let AGH accept requests on all interfaces (see https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#configuration-file; bind_hosts = 0.0.0.0)
- Let AGH also accept DNS requests from the Tailscale subnet 100.64.0.0/10 (see https://tailscale.com/kb/1015/100.x-addresses)
- Configure AGH as custom global nameserver, with 'Override local DNS' enabled, as both the LAN IP and the Tailnet IP (neither work)
- Disable MagicDNS (is this needed?)
- Start Tailscale with the --accept-dns=false parameter, on AGH hosts
I'm rather lost. What do I not see?

**Reply by AkashRajpurohit** [reply]
That's weird, from the steps you have mentioned, I think you have covered everything that is required. I don't have any suggestion on what could be going wrong with your setup here :thinking:

**Reply by camemb3rt** [reply]
- On the Tailscale App, go into Settings (your profile icon) > DNS Settings > Turn off "Use Tailscale DNS"

**Reply by camemb3rt** [reply]
Actually, messing around with it a bit more, try using the local IP address (192.x.x.x) as the global nameserver in the tailscale admin panel instead of the Tailscale IP (100.x.x.x), now DNS rewrites also work.

**Comment by ghost** [comment]
Hi, thanks for your detailed tutorial which i followed and almost got everything working. I just have 1 issue. I have an Asus router then have Adguard and Tailscale installed as per your tutorial. It works okay for a while until my public IP changes then I can no longer access devices from outside the network. Is there a fix for this? Do i need to install another service to update the ip when it changed.

Hope you re able to help.

Thanks

Aidan

**Comment by camemb3rt** [comment]
I was facing the same problem as the other two commenters but I figured out a solution.
- On the Tailscale App, go into Settings (your profile icon) > DNS Settings > Turn off "Use Tailscale DNS"

**Comment by zanetaylor** [comment]
Thanks for this write-up! It was a helpful gut check while moving my existing AdGuard + Tailscale install into a fresh container.
